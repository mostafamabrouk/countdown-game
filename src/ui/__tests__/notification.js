/* eslint-disable no-undef */
import React from "react";
import renderer from "react-test-renderer";
import { Notification } from "../";
import "jest-styled-components";

test("Notification renders withouut any parameters", () => {
  const component = renderer.create(<Notification />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Notification renders with message", () => {
  const component = renderer.create(
    <Notification>This is a message</Notification>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Notification renders with message and error", () => {
  const component = renderer.create(
    <Notification error={true}>This is a message</Notification>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test("Notification renders with message and success", () => {
  const component = renderer.create(
    <Notification success={true}>This is a message</Notification>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
