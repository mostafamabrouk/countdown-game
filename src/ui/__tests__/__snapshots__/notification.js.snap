// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders with message 1`] = `
.c0 {
  padding: 10px;
  background: #eee;
  display: table;
  margin: auto;
}

<p
  className="c0"
/>
`;

exports[`Notification renders with message and error 1`] = `
.c0 {
  padding: 10px;
  background: #eee;
  display: table;
  margin: auto;
  color: #F00;
  background: #FFEBEE;
}

<p
  className="c0"
/>
`;

exports[`Notification renders with message and success 1`] = `
.c0 {
  padding: 10px;
  background: #eee;
  display: table;
  margin: auto;
  color: #6CBC36;
  background: #E8F5E9;
}

<p
  className="c0"
/>
`;

exports[`Notification renders withouut any parameters 1`] = `
.c0 {
  padding: 10px;
  background: #eee;
  display: table;
  margin: auto;
}

<p
  className="c0"
/>
`;
